sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRowConverter.scala

```diff
@@ -436,6 +436,17 @@ private[parquet] class ParquetRowConverter(
           }
         }
 
+      // INT96 timestamp doesn't have a logical type, here we check the physical type instead.
+      case TimestampNTZType if parquetType.asPrimitiveType().getPrimitiveTypeName == INT96 =>
+        new ParquetPrimitiveConverter(updater) {
+          // Converts nanosecond timestamps stored as INT96.
+          // TimestampNTZ type does not require rebasing due to its lack of time zone context.
```

Comment:
Could you tell me why we can treat ltz as ntz directly?

Reply:
* LTZ doesn't really store the time zone info in Parquet files.  Also, Spark uses the long value directly when reading NTZ as LTZ. I am trying to make it simple and Symmetrical.
* If we shift by sessional time zone here, probably we need to do it when reading NTZ as LTZ, which will be a breaking change. Also, the result of NTZ columns will be affected by the sessional time zone conf `spark.sql.session.timeZone`.

Reply:
Got it. Thank you!